.github/actions/sccache/action.yml

```diff
@@ -87,3 +88,50 @@ runs:
         secretKey: ${{ steps.aws-creds.outputs.aws-secret-access-key }}
         sessionToken: ${{ steps.aws-creds.outputs.aws-session-token }}
         use-fallback: false
+
+    - name: (External PR) Install AWS Signer (Linux)
+      if: >
+        runner.os == 'Linux' &&
+        github.event.pull_request.head.repo.full_name != github.event.pull_request.base.repo.full_name &&
+        (contains(runner.name, 'puget') || contains(runner.name, 'mini-'))
+      env:
+        LINUX_SIGNER_URL: https://rolesanywhere.amazonaws.com/releases/1.1.1/X86_64/Linux/aws_signing_helper
+      run: |
+        mkdir -p $HOME/.local/aws
+        curl $LINUX_SIGNER_URL -o $HOME/.local/aws/aws_signing_helper
+        chmod +x $HOME/.local/aws/aws_signing_helper
+      shell: bash
+
+    - name: (External PR) Install AWS Signer (macOS)
+      if: >
+        runner.os == 'macOS' &&
+        github.event.pull_request.head.repo.full_name != github.event.pull_request.base.repo.full_name &&
+        (contains(runner.name, 'puget') || contains(runner.name, 'mini-'))
+      env:
+        DARWIN_SIGNER_URL: https://rolesanywhere.amazonaws.com/releases/1.1.1/X86_64/Darwin/aws_signing_helper
+      run: |
+        mkdir -p $HOME/.local/aws
+        curl $DARWIN_SIGNER_URL -o $HOME/.local/aws/aws_signing_helper
+        chmod +x $HOME/.local/aws/aws_signing_helper
+      shell: bash
+
+    - name: (External PR) Install IAM Anywhere Resources for risc0 Public Cache
+      if: >
+        github.event.pull_request.head.repo.full_name != github.event.pull_request.base.repo.full_name &&
+        (contains(runner.name, 'puget') || contains(runner.name, 'mini-'))
```

Comment:
Hmm, I wonder if we could use a different tag/label for this detection rather than the runner name

Reply:
I made an improvement, but on-prem Linux runners will still require a reference to `runner.name`, at least for now.

- `Benchmark Check` jobs need a way to distinguish between:
  - `self-hosted` runners from AWS (names begin with `i-`, IAM Anywhere setup steps not required) 
  - `self-hosted` on-prem runners (names vary, IAM Anywhere setup steps required)
- If this is not done, jobs running on AWS runners look for the IAM Anywhere stuff, which isn't loaded on there and is not required for the jobs to work.

-----

Fix so on-prem Linux self-hosted runners will pick up the IAM Anywhere jobs, but not AWS ones:
```
if: >
  runner.os == 'Linux' &&
  github.event.pull_request.head.repo.full_name != github.event.pull_request.base.repo.full_name &&
  (runner.environment == 'self-hosted' && (contains(runner.name, 'i-') == 'false'))
```

Fix for on-prem macOS self-hosted runners, which do not need `contains(runner.name,'foo')` since we do not use macOS runners in AWS that could get picked up:
```
if: >
  runner.os == 'macOS' &&
  github.event.pull_request.head.repo.full_name != github.event.pull_request.base.repo.full_name &&
  runner.environment == 'self-hosted'
```

-----

**More info:**
- GitHub [Actions Contexts](https://docs.github.com/en/actions/learn-github-actions/contexts#runner-context) do not currently offer a `runner.label` property to extract and process this through conditions.
- Others have this issue ([example](https://github.com/actions/runner/issues/821)) and have proposed adding this, but nothing yet. 
- There is a [third-party action](https://github.com/marketplace/actions/runner-label-resolver) that will do this, but it requires a Token with permissions at the Org level to pull the labels.